src/main/java/org/mskcc/limsrest/service/PromoteBanked.java

```diff
@@ -503,4 +524,170 @@ public String unaliquotName(String sampleName) {
+                    }
+                }
+            } else {
+                throw new RuntimeException("Could not get service_row for service request id, " +
+                        "check to see if API changed: " + requestId);
+            }
+            log.info("requestId is: " + requestId);
+            log.info("hasCustomForm value: " + hasCustomForm);
+            if (hasCustomForm) {
+                customForms = parseCustomForms(String.format("%s/%s/service_requests/%s/custom_forms.json", baseUrl, core_id_igo, serviceRequestId), restTemplateIGO);
+            }
+            CustomForm customForm = customForms.get(0);
+            log.info("customForm id is:" + customForm.getId());
+            log.info("customForm name is:" + customForm.getName());
+
+            Pattern commentPattern = Pattern.compile("comment", Pattern.CASE_INSENSITIVE);
+            Pattern numOfSamplePattern = Pattern.compile("number of samples", Pattern.CASE_INSENSITIVE);
+            String iLabComment = "";
+            String numOfSamples = "";
+            for (String field : customForm.getFields().keySet()) {
+                log.info("custom form fields are: " + field + customForm.getFields().get(field));
+                Matcher commentMatcher = commentPattern.matcher(field);
+                Matcher numOfSampleMatcher = numOfSamplePattern.matcher(field);
+                boolean commentMatchFound = commentMatcher.find();
+                boolean numOfSamplesMatchFound = numOfSampleMatcher.find();
+                if (commentMatchFound) {
+                    iLabComment = customForm.getFields().get(field);
+                }
+                if(numOfSamplesMatchFound) {
+                    numOfSamples = customForm.getFields().get(field);
+                }
+            }
+            log.info("The comment extracted from iLab request is: " + iLabComment);
+            log.info("num of samples = " + numOfSamples);
+            // Logic related to start and due dates, not used for now
+            String pattern = "MM/dd/yyyy";
+            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern);
+            String date = simpleDateFormat.format(new Date());
+            log.info("start date is: " + date);
+            long timeDiff = 21 * 24 * 60 * 60 * 1000;
+            long time = System.currentTimeMillis() + timeDiff;
+            String dueDate = simpleDateFormat.format(new Date(time));
+            log.info("Due date: " + dueDate);
+
+            String subject = requestId + " (" + numOfSamples + ")";
+            if(iLabComment != null &&  iLabComment.trim().length() > 0) {
+                subject += " #[IGO: iLab Comment]";
+            }
+            if (iLabComment != null)
+                message.setText(iLabComment + " \n#end");
+            message.setSubject(subject);
+
+            Transport.send(message);
```

Comment:
May need to check the return value, I'm not sure sending mail works on igolims in this way?

Reply:
I am now using the workaround we use for pulling iLab into LIMS messaging. 
Now, at promote a text file is being written into  "/pskis34/vialelab/LIMS/AutomatedEmails/teamworkCard/" directory and a cron-job on igo-lims04 is picking text files to send emails to myself and IGO VMB Teamwork board.